src/core/iomgr/tcp_server.h

```diff
@@ -80,4 +80,8 @@ int grpc_tcp_listener_get_port(grpc_tcp_listener *listener);
 void grpc_tcp_listener_ref(grpc_tcp_listener *listener);
 void grpc_tcp_listener_unref(grpc_tcp_listener *listener);
 
+/* potentially (trivially) implementable: */
+int grpc_tcp_listener_get_fd(grpc_tcp_listener *listener, unsigned index);
```

Comment:
It seems like what you are trying for is to have the "head" sibling (is_sibling==0) be the exposed API and the others (is_sibling==1) be hidden.

This is OK as long as you pass the index and the head sibling to tcp_server_posix.c:on_read and on_accept_cb.